packages/libs/eda/src/lib/core/components/computations/plugins/selfCorrelation.tsx

```diff
@@ -0,0 +1,326 @@
+      <H6>What is correlation?</H6>
+      <p>
+        The correlation between two variables (genes, sample metadata, etc.)
+        describes the degree to which their presence in samples co-fluctuate.
+        For example, the Age and Shoe Size of children are correlated since as a
+        child ages, their feet grow.
+      </p>
+      <p>
+        Here we look for correlation between the abundance of different taxa at
+        a given taxonomic level
+      </p>
+      <br></br>
+      <H6>Inputs:</H6>
+      <p>
+        <ul>
+          <li>
+            <strong>Taxonomic Level.</strong> The taxonomic abundance data to be
+            used in the calculation.
+          </li>
+          <li>
+            <strong>Method.</strong> The type of correlation to compute. The
+            Pearson method looks for linear trends in the data, while the
+            Spearman method looks for a monotonic relationship. For Spearman and
+            Pearson correlation, we use the rcorr function from the Hmisc
+            package. The SparCC method is a compositional correlation method
+            appropriate for taxonomic abundance data and any other compositional
+            data.
+          </li>
+          <li>
+            <strong>Prevalence Prefilter.</strong> Remove variables that do not
+            have a set percentage of non-zero abundance across samples. Removing
+            rarely occurring features before calculating correlation can prevent
+            some spurious results.
+          </li>
+        </ul>
+      </p>
+      <br></br>
+      <H6>Outputs:</H6>
+      <p>
+        For each pair of variables, the correlation computation returns
+        <ul>
+          <li>
+            Correlation coefficient. A value between [-1, 1] that describes the
+            similarity of the input variables. Positive values indicate that
+            both variables rise and fall together, whereas negative values
+            indicate that as one rises, the other falls.
+          </li>
+          <li>
+            P Value. A measure of the probability of observing the result by
+            chance.
+          </li>
+        </ul>
+      </p>
+      <br></br>
+      <H6>More Questions?</H6>
+      <p>
+        Check out the{' '}
+        <a href="https://github.com/VEuPathDB/microbiomeComputations/blob/master/R/method-correlation.R">
+          correlation function
+        </a>{' '}
+        in our{' '}
+        <a href="https://github.com/microbiomeDB/MicrobiomeDB">
+          microbiomeComputations
+        </a>{' '}
+        R package.
+      </p>
+    </div>
+  );
+
+  const correlationMethodSelectorText = useMemo(() => {
+    if (configuration.correlationMethod) {
+      return (
+        CORRELATION_METHODS.find(
+          (method) => method.value === configuration.correlationMethod
+        )?.displayName ?? 'Select a method'
+      );
+    } else {
+      return 'Select a method';
+    }
+  }, [configuration.correlationMethod]);
```

Comment:
I would lose the `useMemo` here. The return value of the callback is a string, and the same string values will always be referentially equal. Also, the `CORRELATION_METHODS` array is small, so traversing it will not have a large impact on performance.